contrib/IECoreUSD/src/IECoreUSD/PrimitiveAlgo.cpp

```diff
@@ -432,6 +439,26 @@ void IECoreUSD::PrimitiveAlgo::readPrimitiveVariables( const pxr::UsdGeomPrimvar
 			continue;
 		}
 
+		// skip custom constant prim vars not tag with special Cortex metadata
+		if ( primVar.GetAttr().IsCustom() )
```

Comment:
I was just getting Gaffer set up with the PXR_PLUGINPATH_NAME stuff so that the metadata could be registered OK, and to my surprise it was completely unnecessary for keeping constant primitive variables round-tripping, even though the metadata fails to write with this error :

```Unregistered metadata field: IECOREUSD_CONSTANT_PRIMITIVE_VARIABLE```

This turns out to be because the `IsCustom()` check here prevents us from even considering the metadata on read. Is there a reason we're checking `IsCustom()` here?

Reply:
That makes sense to me that the round tripping of Constant PrimVar would still works even without the registration of the custom metadata.

The purpose of the custom metadata is to avoid "loading" every since Custom Constant PrimVar as a Cortex Attribute.

Regarding using `IsCustom` here, when I was working on this, I had a print that was showing the metadata test performed on non custom PrimVars like Cs or primvars:st so I decided to use `IsCustom` to prevent testing the metadata on PrimVars that was "guaranteed" not to have it.

Perhaps the problem could be that we are not creating PrimVars as custom by default when we probably should?

Reply:
> The purpose of the custom metadata is to avoid "loading" every since Custom Constant PrimVar as a Cortex Attribute.

By "custom metadata" do you mean `cortexPrimitiveVariableMetadataToken` or `custom`? I thought the purpose of `cortexPrimitiveVariableMetadataToken` was to allow us to round-trip constant Cortex primitive variables through USD without them becoming Cortex attributes on read. Nobody else will be authoring `cortexPrimitiveVariableMetadataToken`, and the way we're authoring it currently, it won't be checked on read at all.

Is it actually our intent that non-`custom` primvars should be loaded as primitive variables rather than attributes in Cortex? If so, why? How does `custom` change the USD semantics of inheritance for `primvars:`?

> Perhaps the problem could be that we are not creating PrimVars as custom by default when we probably should?

I've seen no documentation to suggest that `custom` is required to be applied to primvars, but maybe I missed something? I assumed that the PrimvarsAPI was the true way of dealing with primvars.

Reply:
After talking to Daniel, I understand that I was confused between USD Attribute that can be `custom` and USD PrimVars that are never.

I will fix the PR to account for that.
Sorry for the confusion